faunadb-java/src/main/java/com/faunadb/client/types/time/HighPrecisionTime.java

```diff
@@ -0,0 +1,133 @@
+package com.faunadb.client.types.time;
+
+import org.joda.time.Instant;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.String.format;
+import static java.util.Objects.hash;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Wraps an instance of {@link Instant} and adds micro and nanosecond precision to it.
+ */
+public class HighPrecisionTime {
```

Comment:
this should PROBABLY be comparable / extend compareTo? Doubt we need it but someone sometime may want to compare these things